pkg/publicapi/client/v2/options.go

```diff
@@ -37,11 +40,47 @@ type Options struct {
 
 	// Headers is a map of headers to add to all requests.
 	Headers http.Header
+
+	// TLSOptions provides info on how we want to use TLS
+	TLS TLSOptions
+}
+
+type TLSOptions struct {
```

Comment:
I think this, and the above `Optons` structure should really be named `Config*` (as you have rightly called the variable of this type in some places), the functions that mutate it are the actual _options_. 

Additionally the `OptionsFn` should probably be made to return errors so we can avoid panicking - I know you are just building on this and didn't introduce the change, but if it's not too much trouble I'd like to see this debt paid down as a part of this work.

Generally is abnormal (imho) usage of [functional options pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis)...